packages/dcos-integration-test/extra/conftest.py

```diff
@@ -463,17 +463,19 @@ def _pool_for_marathon_app(app_id):
                     'Application deployment failed, reason: {}'.format(data['app']['lastTaskFailure']['message'])
                 )
 
-            if (
-                data['app']['tasksRunning'] == app_definition['instances'] and
-                (not check_health or data['app']['tasksHealthy'] == app_definition['instances'])
-            ):
+            if (data['app']['tasksRunning'] != app_definition['instances']):
+                logging.info('Waiting for application to be deployed: ' \
+                             'Not all instances are running: {}'.format(repr(data)))
+                return None
+            elif (check_health and data['app']['tasksHealthy'] != app_definition['instances']):
+                logging.info('Waiting for application to be deployed: ' \
+                             'Not all instances are healthy: {}'.format(repr(data)))
+                return None
+            else:
```

Comment:
could we make this logic so that we only return / get here if we find things to be healthy explicitly?

I don't like having the fall through of "we didn't see anything was unhealthy" -> "everything is okay".